tools/harness-thci/OpenThread.py

```diff
@@ -780,37 +768,67 @@ def getMAC(self, bType=MacType.RandomMac):
                 macAddr64 = self.__sendCommand('joiner id')[0]
             else:
                 macAddr64 = self.__sendCommand('extaddr')[0]
-        print macAddr64
+        print(macAddr64)
 
         return int(macAddr64, 16)
 
     def getLL64(self):
         """get link local unicast IPv6 address"""
-        print '%s call getLL64' % self.port
-        return self.__getIp6Address('link local')
-
-    def getMLEID(self):
-        """get mesh local endpoint identifier address"""
-        print '%s call getMLEID' % self.port
-        return self.__getIp6Address('mesh EID')
+        print('%s call getLL64' % self.port)
+        return self.__sendCommand('ipaddr linklocal')[0]
 
     def getRloc16(self):
         """get rloc16 short address"""
-        print '%s call getRloc16' % self.port
+        print('%s call getRloc16' % self.port)
         rloc16 = self.__sendCommand('rloc16')[0]
         return int(rloc16, 16)
 
     def getRloc(self):
-        """get router locator unicast IPv6 address"""
-        print '%s call getRloc' % self.port
-        return self.__getIp6Address('rloc')
+        """get router locator unicast Ipv6 address"""
+        print('%s call getRloc' % self.port)
+        return self.__sendCommand('ipaddr rloc')[0]
 
-    def getGlobal(self):
+    def __getGlobal(self):
         """get global unicast IPv6 address set
            if configuring multiple entries
         """
-        print '%s call getGlobal' % self.port
-        return self.__getIp6Address('global')
+        print('%s call getGlobal' % self.port)
+        fullIpAddrs = []
+        globalAddrs = []
+        rlocAddr = None
+
+        addrs = self.__sendCommand('ipaddr')
+
+        # convert address to lower case in full expression, and find rloc address as
+        # a reference for current mesh local prefix as for some TCs, mesh local prefix
+        # may be updated through pending dataset management.
+        for ip6Addr in addrs:
+            if ip6Addr == 'Done':
+                break
+
+            fullIp = ModuleHelper.GetFullIpv6Address(ip6Addr).lower()
+            fullIpAddrs.append(fullIp)
+
+            if fullIp.startswith('fd00') and fullIp.startswith('0000:00ff:fe00:', 20) and \
+                    not fullIp.startswith('fc', 35):
+                rlocAddr = fullIp
```

Comment:
What about to use `self.getRloc()` before this loop instead?

Reply:
yes. just updated.